File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int , int>mp ;
        for(int num : nums)
        {
            mp[num]++; 
            if(mp[num]  > 1)
                return true ;     
        }
        return false ;
    }
};

File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> bool:

        left = 0 
        right = len(nums)-1 

        while left <= right :
            middle = left + ((right - left) // 2)
            
            if nums[middle] == target :
                return True  
            elif nums[middle] < target :
                left = middle + 1
            else : 
                right = middle - 1

        return False
        
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        for i in range(len(matrix)) :
            if self.search(matrix[i] , target) :
                return True 
        
        return False 

      